website/content/docs/auth/login-mfa.mdx

```diff
@@ -0,0 +1,170 @@
+like a one-time passcode, before being authenticated.
+
+There are two ways to validate a login request that is subject to MFA validation.
+
+ ### Single-Phase Login
+ In the Single-phase login, the required MFA information is embedded in a login request using
+ the `X-VAULT-MFA` header. In this case, the MFA validation is done
+ as a part of the login request.
+
+ MFA credentials are retrieved from the `X-Vault-MFA` HTTP header. The format of
+ the header is `mfa_method_id[:key[=value]]`. The item in the `[]` is optional. If there are more than
+ one MFA method that needs to be validated, a user can pass in multiple `X-Vault-MFA` HTTP headers.
+
+ #### Sample Request
+
+ ```shell-session
+ $ curl \
+     --header "X-Vault-Token: ..." \
+     --header "X-Vault-MFA:d16fd3c2-50de-0b9b-eed3-0301dadeca10:695452" \
+     http://127.0.0.1:8200/v1/auth/userpass/login/alice
+ ```
+
+ ### Two-Phase Login
+ The more conventional, and prevalent MFA method is a two request mechanism, also referred to as Two-phase Login MFA.
+ In Two-phase login the `X-Vault-MFA` header is not provided in the request. In this case, after sending a regular login request,
+ the user receives an auth response in which MFA requirements are included. MFA requirements contain an MFA request ID
+ which identifies the login request that needs to get validated. In addition, MFA requirements contain MFA constraints
+ that identify which MFA types should be used to validate the request, the corresponding method IDs, and
+ a boolean value showing whether the MFA method uses passcodes or not. MFA constraints form a nested map in MFA Requirement,
+ and represent all MFA enforcements that match a login request.
+
+#### Sample Two-Phase Login Response
+
+```json
+{
+  "request_id": "1044c151-13ea-1cf5-f6ed-000c42efd477",
+  "lease_id": "",
+  "lease_duration": 0,
+  "renewable": false,
+  "data": null,
+  "warnings": [
+    "A login request was issued that is subject to MFA validation. Please make sure to validate the login by sending another request to mfa/validate endpoint."
+  ],
+  "auth": {
+    "client_token": "",
+    "accessor": "",
+    "policies": null,
+    "token_policies": null,
+    "identity_policies": null,
+    "metadata": null,
+    "orphan": false,
+    "entity_id": "",
+    "lease_duration": 0,
+    "renewable": false,
+    "mfa_requirement": {
+      "mfa_request_id": "d0c9eec7-6921-8cc0-be62-202b289ef163",
+      "mfa_constraints": {
+        "enforcementConfigUserpass": {
+          "any": [
+            {
+              "type": "totp",
+              "id": "820997b3-110e-c251-7e8b-ff4aa428a6e1",
+              "uses_passcode": true
+            }
+          ]
+        }
+      }
+    }
+  }
+}
+```
+
+ Note that the `uses_passcode` boolean value is always set to true for TOTP, and always set to false for Okta and PingID.
+ For DUO method, the value can be configured as part of the method configuration.
```

Comment:
In some places, I see `DUO` and others `Duo`. Duo's web site uses `Duo`, so maybe we could use that everywhere?